### DISCUSSION TITLE
How to declare a job as busy or finished?

### DISCUSSION THREAD

**Discussion by smit-luvani** [Hexagon/croner#234] [Q&A] [answered]
I saw isBusy() returns true/false. How can I tell programmatically that this job is currently running and will finish when an internal callback responds? 
   
```
const Job = Cron('*/30 * * * *', {}, (job, context) => {
    longRunningFunction().then((result) => {
      // job finished
   }).catch(error => {
     // job finished
   })
});
```

What if I want to check after 2 minutes that the job is still busy or not?

**Comment by Hexagon** [comment, marked as answer]
IsBusy() should be true as long as the function is still running, or the promise returned by the function isn't fulfilled. Try to declare the triggered function async, and await the promise before returning. Or just return the promise.
